components/powersensor/server_test.go

```diff
@@ -0,0 +1,140 @@
+package powersensor_test
+
+import (
+	"context"
+	"errors"
+	"testing"
+
+	pb "go.viam.com/api/component/powersensor/v1"
+	"go.viam.com/test"
+
+	"go.viam.com/rdk/components/powersensor"
+	"go.viam.com/rdk/components/sensor"
+	"go.viam.com/rdk/resource"
+	"go.viam.com/rdk/testutils/inject"
+)
+
+var (
+	workingPowerSensorName = "ps1"
+	failingPowerSensorName = "ps2"
+	missingPowerSensorName = "ps3"
+	errVoltageFailed       = errors.New("can't get voltage")
+	errCurrentFailed       = errors.New("can't get current")
+	errPowerFailed         = errors.New("can't get power")
+	errPowerSensorNotFound = errors.New("not found")
+)
+
+func newServer() (pb.PowerSensorServiceServer, *inject.PowerSensor, *inject.PowerSensor, error) {
+	workingPowerSensor := &inject.PowerSensor{}
+	failingPowerSensor := &inject.PowerSensor{}
+	powerSensors := map[resource.Name]powersensor.PowerSensor{
+		powersensor.Named(workingPowerSensorName): workingPowerSensor,
+		powersensor.Named(failingPowerSensorName): failingPowerSensor,
+	}
+
+	powerSensorSvc, err := resource.NewAPIResourceCollection(sensor.API, powerSensors)
+	if err != nil {
+		return nil, nil, nil, err
+	}
+
+	return powersensor.NewRPCServiceServer(powerSensorSvc).(pb.PowerSensorServiceServer), workingPowerSensor, failingPowerSensor, nil
```

Comment:
This is 130ish characters long, and does at least 2 things (create a server and return a bunch of stuff). I'd split it up into 2 shorter lines that do 1 thing each.